lib/linux/sys/proctable.rb

```diff
@@ -169,6 +171,10 @@ def self.ps(pid=nil)
         # Get control groups to which the process belongs
         struct.cgroup = IO.readlines("/proc/#{file}/cgroup").map { |l| CgroupEntry.new(l) } rescue []
 
+        # Read smaps, return a parseable empty string if we don't have permissions
+        smaps_contents = IO.read("/proc/#{file}/smaps") rescue ""
```

Comment:
This is going to be a lot of rescued exceptions if not run with appropriate privileges. I wonder if checking with File.readable? first would be faster. Can you bench?